src/flat_bvh.rs

```diff
@@ -41,6 +43,14 @@ pub struct FlatNode<T: BHValue, const D: usize> {
     pub shape_index: u32,
 }
 
+impl<T: BHValue, const D: usize> FlatNode<T, D> {
+    /// Returns whether the node is a leaf node.
+    #[must_use]
+    fn is_leaf(&self) -> bool {
+        self.entry_index == u32::MAX
+    }
```

Comment:
I'd prefer `pub fn`, as long as the rest of the `FlatNode` internals are public.